src/modules/mc_pos_control/PositionControl/PositionControl.cpp

```diff
@@ -185,15 +185,15 @@ void PositionControl::_velocityControl(const float dt)
 	// Use tracking Anti-Windup for horizontal direction: during saturation, the integrator is used to unsaturate the output
 	// see Anti-Reset Windup for PID controllers, L.Rundqwist, 1990
 	const Vector2f acc_sp_xy_produced = Vector2f(_thr_sp) * (CONSTANTS_ONE_G / _hover_thrust);
-	const float arw_gain = 2.f / _gain_vel_p(0);
 
 	// The produced acceleration can be greater or smaller than the desired acceleration due to the saturations and the actual vertical thrust (computed independently).
 	// The ARW loop needs to run if the signal is saturated only.
-	const Vector2f acc_sp_xy = _acc_sp.xy();
-	const Vector2f acc_limited_xy = (acc_sp_xy.norm_squared() > acc_sp_xy_produced.norm_squared())
-					? acc_sp_xy_produced
-					: acc_sp_xy;
-	vel_error.xy() = Vector2f(vel_error) - arw_gain * (acc_sp_xy - acc_limited_xy);
+	if (_acc_sp.xy().norm_squared() > acc_sp_xy_produced.norm_squared()) {
+		const float arw_gain = 2.f / _gain_vel_p(0);
```

Comment:
Side note, shouldn't this be split X/Y?

Reply:
I was wondering what happens currently if _acc_sp.xy() and acc_sp_xy_produced.norm_squared are pointing in opposite directions. Is it possible for this to happen and for the length of _acc_sp.xy() to be greater than acc_sp_xy_produced.norm_squared? I can't think of a way for this to happen. 

However the ARW algorithm should work with x and y split because it acts over a time constant and fades out the excess integral proportional to the amount of saturation. It would be worth a test.

@MaEtUgR what do you think?

Reply:
> what happens currently if _acc_sp.xy() and acc_sp_xy_produced.norm_squared are pointing in opposite directions

I don't think this is possible as the accel saturation can't change the sign of the desired acceleration.

> However the ARW algorithm should work with x and y split because it acts over a time constant and fades out the excess integral proportional to the amount of saturation. It would be worth a test.

Yes, we can split it if needed.